### PR TITLE
Added .value_name() call

### PR DESCRIPTION
**Pull request by uomoinverde**
Ok, re-PR the whole thing. I've learnt a lot, first of all: learn to use your things before actually doing stuff. Thank you for letting me contribute. :-)

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Awesome, thank you very much!

File: imag-store/src/ui.rs [hunk 1]
```diff
@@ -86,6 +93,7 @@ pub fn build_ui<'a>(app: App<'a, 'a>) -> App<'a, 'a> {
                                     .short("w")
                                     .takes_value(true)
                                     .help("Filter with 'header.field=foo' where the header field 'header.field' equals 'foo'")
+                                    .value_name("header.field=foo")
```
Comment (matthiasbeyer) [inline imag-store/src/ui.rs]: This would be better named `"equation"` or `"spec"`.

File: imag-tag/src/ui.rs [hunk 2]
```diff
@@ -58,7 +62,8 @@ pub fn build_ui<'a>(app: App<'a, 'a>) -> App<'a, 'a> {
                         .short("s")
                         .takes_value(true)
                         .required(false)
-                        .help("Seperated by string"))
+                        .help("Separated by string")
+                        .value_name("SEP"))
```
Comment (matthiasbeyer) [inline imag-tag/src/ui.rs]: I'd prefer "`SEPERATOR"`.

**Comment by matthiasbeyer** [comment]
Two more things: 

1. Why did you remove the blank lines at the end of the files?
2. Why did you squash all commits into one commit?

---

If you want to learn some more things on how to use git, we can fix this within this PR. Just tell me if you want to. If you do not care, I'll merge as proposed.

---

Also: Please tell me _all the things_ I just told you on how to do PR you didn't find in the CONTRIBUTING file, so I can add the things. Feel free to open an issue for this! This way I can improve the CONTRIBUTING file so everyone can read how I'd like to have the PRs :smile:

File: imag-store/src/ui.rs (hunk 1 above)
Comment (uomoinverde) [inline imag-store/src/ui.rs]: What about `"filter"`? I'll wait for your final response before changing this.

File: imag-tag/src/ui.rs (hunk 2 above)
Comment (uomoinverde) [inline imag-tag/src/ui.rs]: Ok. Would `"SEPARATOR"` be better?

**Comment by uomoinverde** [comment]
To answer your questions:
   1. I think Atom did that, it wasn't my intentions (I can add them again); 
   2. Fact is, I got things messed in my fork and I decided to make a new one. I re-edited (credits to my brain memory in this case) and commited just once.

---

I'd love to learn more. I'm going to read the [book](https://git-scm.com/book/en/v2), but I'd also like to be guided so if you don't mind, tell me what to do! :-)

---

I'll take a look at CONTRIBUTING. Thank you for the information!

File: imag-store/src/ui.rs (hunk 1 above)
Comment (matthiasbeyer) [inline imag-store/src/ui.rs]: `"filter"` sounds good as well, yes!

File: imag-tag/src/ui.rs (hunk 2 above)
Comment (matthiasbeyer) [inline imag-tag/src/ui.rs]: Ah, yes. Typo on my part, sorry.

**Comment by matthiasbeyer** [comment]
> I think Atom did that, it wasn't my intentions (I can add them again);

Yes, that would be really nice!

> Fact is, I got things messed in my fork and I decided to make a new one. I re-edited (credits to my brain memory in this case) and commited just once.

If you care to split them up into several patches (one per file for example), see below.

> I'd love to learn more. I'm going to read the book, but I'd also like to be guided so if you don't mind, tell me what to do! :-)

Okay, nice!

So for the fixup things, create your fixups and commit them with `git commit --amend --no-edit`, so they are put into the latest commit without changing the commit message.

For the commit-split, do the following:

1. Make sure you checked out your branch (`git checkout add-value-name-fields`)
2. Do `git rebase -i origin/master`
3. A Text editor will open. Change the `"pick"` at the beginning of the line to `"e"` or `"edit"`. The rebase will then halt at the commit (before committing) and you can then edit the commit.
4. If the upper thing worked, do `git reset` (NO FLAGS!!!), the files which are staged will now be unstaged, so you can commit them one by one
5. Commit each file as usual (`git add <file>; git commit <file> -m '<message>'`)
6. After you committed all the files do `git rebase --contine`.
7. Have a look at the history (`git log --oneline --all --graph --color --decorate`) if it looks good, do a `git push add-value-name-fields --force`).

Note: You are allowed to do a forced push here because you are working on a pull request. **NEVER EVER** do a forced push on a branch where others already based work on (for example a master branch or something like this).

Feel free to ask questions! :small_blue_diamond: :smile:

**Comment by uomoinverde** [comment]
I'm trying to following your instructions, but I can't get my head off step 4. Files don't get unstaged. :disappointed:
I edited the `rebase-todo` changing the flag from *pick* to *edit*, but I didn't anything from step 3 to 4...could this be the matter?

**Comment by matthiasbeyer** [comment]
Ok, alternative appoach.

First of all, your branch has conflicts. Do you know how to resolve them? If not, I can do these things for you, if you want me to.

If you want to do it yourself, feel free to do so.

After that, you can try to `git reset --soft origin/master && git reset` and then commit each change on its own.

As said: If you don't want to do these things, I can help you by doing it, you can then push the changes I made to this PR and we are good.

**Comment by uomoinverde** [comment]
I'd really love to learn by doing these! But I'm slowing down this relatively small change to `imag` that should have been completed ~1 week ago. So, feel free to help me by doing them and I'll do the push...next time, I'll keep in mind all the things you told me, adhere to the guidelines and do a better job. Promised :-)

**Comment by matthiasbeyer** [comment]
Hey, that's absolutely no problem! If you want to try, go ahead, really! IMHO it is more important to learn than to have progress! So absolutely no problem if it takes some time for you to understand these things, really!

Feel free to move the conversation about how to do these things to email: mail [at] beyermatthias (dot) de so we do not spam in here with this conversation.

I'm glad to help! (Though I might be in the swimming pool until evening today :smile:

**Comment by matthiasbeyer** [comment]
Closed after 18 days of inactivity and superceded by #466. Would still be nice to see you contributing again, @uomoinverde !

**Pull request closed without merging**